server/server.js

```diff
@@ -1,15 +1,73 @@
+const PORT = 4500;
 const express = require("express");
+const cors = require("cors");
+const morgan = require("morgan");
+
 const app = express();
-const port = process.env.PORT || 5000;
+const fs = require("fs");
+// const fs = require("fs").promises;
+
+app.use(cors());
+app.use(morgan("dev"));
 
-app.listen(port, () => console.log(`Listening on port ${port}`));
+app.use(express.static("public"));
+app.use(express.json());
+app.use(express.urlencoded({ extended: true }));
 
 // Store and retrieve your videos from here
 // If you want, you can copy "exampleresponse.json" into here to have some data to work with
-let videos = [];
+// let videos = [];
+
+function getConvertedDataFromBookingsJSON() {
+  const rawData = fs.readFileSync("./videos.json");
+  const objData = JSON.parse(rawData);
+  return objData;
+}
+
+function writeUpdateDataToJsonFile(data) {
+  const jsonFile = "./videos.json";
+  const newData = JSON.stringify(data);
+  fs.writeFile(jsonFile, newData, (err) => {
+    if (err) {
+      console.error(err);
+    } else {
+      console.log("File written successfully.");
+    }
+  });
+}
+
+// app.use((req, res, next) => {
+//   res.header("Access-Control-Allow-Origin", "*"); // Allow requests from any domain
+//   next();
+// });
 
-// GET "/"
 app.get("/", (req, res) => {
-  // Delete this line after you've confirmed your server is running
   res.send({ express: "Your Backend Service is Running" });
 });
+
+app.get("/video", (req, res) => {
+  return res.status(200).send(getConvertedDataFromBookingsJSON());
+});
+
+app.post("/video", (req, res) => {
+  const { title, url } = req.body;
+  if (!title && !url) {
+    res.status(400).json({ error: "One or several filds wasn't filed" });
+  }
+
+  const objData = getConvertedDataFromBookingsJSON();
+  objData.push({
+    ///////////////////////////////////////////////
+    id: req.body.url.split("=")[1],
```

Comment:
What if the URL didn't contain an `=`? What would happen then? How can we handle that?

Reply:
Good point, I have fixed that in my new commit. 
When a user makes a PUSH request, it sends the corrected URL controlled inside the handleSubmit, which makes sure that it's passing the fixed format of URL.